### PR TITLE
Fix flaky globalrole test

### PR DESCRIPTION
**Pull request by rmweir**
Problem:
Globalrole test results have been flakey. This is not due to the test. The validator is using a lister which does not always update quickly enough after object creation, leading to not found errors.

Solution:
Use client/interface instead of a lister.

Issue:
https://github.com/rancher/rancher/issues/24115

### CODE REVIEW

File: pkg/api/customization/globalrole/validator.go [hunk 1]
```diff
@@ -5,18 +5,19 @@ import (
...
+	GlobalRoleClient v3.GlobalRoleInterface
 }
 
 func (w Wrapper) Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
 	if request.Method != http.MethodPut {
 		return nil
 	}
 
-	gr, err := w.GlobalRoleLister.Get("", request.ID)
+	gr, err := w.GlobalRoleClient.Get(request.ID, v1.GetOptions{})
```
Comment (kinarashah) [inline pkg/api/customization/globalrole/validator.go]: do you think it'd make sense to first check Lister and use client only if errors isNotFound?

Comment (dramich) [reply pkg/api/customization/globalrole/validator.go]: This should go back to using the lister. The test is doing something atypical for how a grb is going to be used.

**Comment by dramich** [comment]
This is a test issue. If the error returned is a 404 then it's known the lister isn't updated yet so the test can wait and try again.

**Comment by dramich** [comment]
lgtm

**Pull request merged**